source/Calamari.Shared/Integration/Scripting/CombinedScriptEngine.cs

```diff
@@ -29,13 +29,15 @@ public CombinedScriptEngine(IEnumerable<IScriptWrapper> scriptWrapperHooks)
             this.scriptWrapperHooks = scriptWrapperHooks;
         }
 
+
         public ScriptSyntax[] GetSupportedTypes()
         {
-            return (CalamariEnvironment.IsRunningOnNix || CalamariEnvironment.IsRunningOnMac)
-                ? new[] { ScriptSyntax.Bash, ScriptSyntax.PowerShell, ScriptSyntax.CSharp, ScriptSyntax.FSharp, ScriptSyntax.Python }
-                : new[] { ScriptSyntax.PowerShell, ScriptSyntax.CSharp, ScriptSyntax.FSharp, ScriptSyntax.Python };
-        }
+            var preferredScriptSyntax = new [] { ScriptSyntaxHelper.GetPreferredScriptSyntaxForEnvironment() };
+            var commonScriptSyntax =  new[] { ScriptSyntax.PowerShell, ScriptSyntax.CSharp, ScriptSyntax.FSharp, ScriptSyntax.Python };
```

Comment:
You might rename this to be even more descriptive: `scriptSyntaxesSupportedOnAllPlatforms`